### PR TITLE
Added Dockerfiles and Actions

### PR DESCRIPTION
**Pull request by DhruvKoolRajamani**
Verified base dockerfile [dockerfile](./Dockerfile) for Github Action (Automated Build). Added dockerfiles to dockerhub with added base image of tensorflow and tensorflow-gpu for containerized training. Containers can also be pulled from singularity to run on the High Performance Cluster in WPI.

**Summary**

- [x] Github Action - On push and pull request
- [x] Dockerhub dockerfiles.
- [x] Added apt-requirements and pip-requirements for detailed installation on fresh Ubuntu installs.

**To Do**

- [ ] Test port forwarding for topics.
- [ ] Test Camera topics for visual feedback during training/simulation runs.

### CODE REVIEW

**Comment by DhruvKoolRajamani** [comment]
This can be merged, all changes have been tested and automated builds have also successfully completed on my fork. The latest dockerfiles are also available under [dockerhub](https://hub.docker.com/repository/docker/dhruv7396/ambf).

Working on testing this with Singularity to allow @VigneshManoj and @cshreyastech to run their training on the High Performance Compute cluster.

**Comment by adnanmunawar** [comment]
Thanks for all the great work and for the exciting additions. I shall test it out soon and merge it.

**Comment by DhruvKoolRajamani** [comment]
I also added details on how to run the file in the README, will modify it once I get time to test port forwarding so you can access the topics being published.

**Comment by cshreyastech** [comment]
Thanks @DhruvKoolRajamani

File: ambf_ros_modules/ambf_comm/scripts/ambf_client.py [hunk 1]
```diff
@@ -115,6 +121,9 @@ def get_obj_handle(self, a_name):
         obj = self._objects_dict.get(a_name)
         if obj:
             obj.set_active()
+            obj.set_children_name_published_state(True)
+            obj.set_joint_name_published_state(True)
+            obj.set_joint_pos_published_state(True)
```
Comment (cshreyastech) [inline ambf_ros_modules/ambf_comm/scripts/ambf_client.py]: Do we want to first check the existence of children before enabling?

**Comment by cshreyastech** [comment]
@DhruvKoolRajamani @adnanmunawar I tested IK of PSM with python2 and python3. Both worked but provided slightly a different number.

**Pull request merged**